### PR TITLE
fix(NODE-3021): fix a long standing bug in Decimal128.fromString()

### PR DESCRIPTION
**Pull request by tzhuan**
## Description

Fix a long standing bug in `Decimal128.fromString()`

  - https://github.com/mongodb/js-bson/issues/334
  - https://jira.mongodb.org/browse/NODE-3021

**What changed?**

The `representation` string was wrongly used to find the significant digits. The `digits` array should be used instead.

Note that the v2.0 and v3.0 branches also have this bug.

### CODE REVIEW

File: test/node/decimal128_tests.js [hunk 1]
```diff
@@ -1210,6 +1210,8 @@ describe('Decimal128', function () {
     expect(new Decimal128('00').toString()).to.equal('0');
     expect(new Decimal128('0.5').toString()).to.equal('0.5');
     expect(new Decimal128('-0.5').toString()).to.equal('-0.5');
+    expect(new Decimal128('-0.0097').toString()).to.equal('-0.0097');
+    expect(new Decimal128('-0.0011').toString()).to.equal('-0.0011');
```
Comment (dariakp) [inline test/node/decimal128_tests.js:1214]:
```suggestion
    expect(new Decimal128('-0.0011').toString()).to.equal('-0.0011');
    expect(new Decimal128('-0.00110').toString()).to.equal('-0.00110');
    expect(new Decimal128('0.0011').toString()).to.equal('0.0011');
    expect(new Decimal128('0.00110').toString()).to.equal('0.00110');
```

**Review by dariakp — CHANGES_REQUESTED** [review]
Thank you for submitting this fix, I just have a suggestion for a few more test cases around the leading/trailing zero

**Review by nbbeeken — APPROVED** [review]
LGTM including the suggested tests, Thanks again!

**Review by dariakp — APPROVED** [review]

**Comment by tzhuan** [comment]
Thank you!

Hope that you can make a new bson npm release (and also a new mongodb node driver npm release) as soon as possible because this bug is potentially a significant vulnerability especially for financial usages.

**Comment by dariakp** [comment]
@tzhuan Yes, we are definitely planning to do a release soon. Thank you again!
